### PR TITLE
fix(DB/gameobject): Improve Terokkar Forest gathering nodes

### PR DESCRIPTION
**Pull request by Gultask**
I still need to find open unused GUIDs and do some preliminary testing. I can't seem to be able to repopulate my DB so I'm out of a local for now.

## Changes Proposed:
-  Replaces MASTER pools
-  Introduces 37 new sniffed ore nodes
- Creates pool_pools for Khorium and Rich Adamantite
- 2% for Fel Iron Khorium, 3% for Adamantite Khorium and 4% for Rich Adamantite Khorium
- Rich Adamantite has 30% chance
- See comment for more info

## Issues Addressed:

- Closes 

## SOURCE:

WotLK Classic for numbers, Retail for the sniffs

## Tests Performed:

- 1 pass, not tested respawn yet.

## How to Test the Changes:

1. Use Gathermate/Gatherer
2. Tap the nodes (don't pick them yet!)
3. Count all spawns, compare to the images below, see if they correlate
4. Pick them. See if it respawns later

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
> I can't seem to be able to repopulate my DB so I'm out of a local for now.

Please fetch the upstream of the pr.

File: data/sql/updates/pending_db_world/rev_1666109692456128300.sql [hunk 1]
```diff
@@ -0,0 +1,1310 @@
...
+(@POOLMOTHER+2 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 3 '),
+(@POOLMOTHER+3 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 4 '),
+(@POOLMOTHER+4 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 5 '),
+(@POOLMOTHER+5 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 6 '),
+(@POOLMOTHER+6 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 7 '),
+(@POOLMOTHER+7 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 8 '),
+(@POOLMOTHER+8 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 9 '),
+(@POOLMOTHER+9 , 1, 'Terokkar Forest - Fel Iron Deposit - Group 10'),
+(@POOLMOTHER+10, 1, 'Terokkar Forest - Fel Iron Deposit - Group 11'),
+(@POOLMOTHER+11, 1, 'Terokkar Forest - Fel Iron Deposit - Group 12'),
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1666109692456128300.sql:25]:
```suggestion
(@POOLMOTHER+11, 1, 'Terokkar Forest - Fel Iron Deposit - Group 12');
```

File: data/sql/updates/pending_db_world/rev_1666109692456128300.sql [hunk 2]
```diff
@@ -0,0 +1,1310 @@
...
+(@GUID+103, 181557, 530, 3519, 0, 1, 1, -3306.21, 6077.18, 2.40368, 2.05949, 0, 0, 0.857167, 0.515038, 3600, 255, 1, '', 0),
+(@GUID+105, 181557, 530, 3519, 0, 1, 1, -3400.92, 5671.56, 3.85431, 0.174533, 0, 0, 0.0871557, 0.996195, 3600, 255, 1, '', 0),
+(@GUID+107, 181557, 530, 3519, 0, 1, 1, -3300.47, 5586.64, -9.90915, 5.84685, 0, 0, -0.216439, 0.976296, 3600, 255, 1, '', 0),
+(@GUID+109, 181557, 530, 3519, 0, 1, 1, -3468.94, 5426.26, -3.36355, 0, 0, 0, 0, 0, 3600, 255, 1, '', 0),
+(@GUID+111, 181557, 530, 3519, 0, 1, 1, -2975.86, 5724.72, 4.78636, 4.85202, 0, 0, -0.656058, 0.75471, 3600, 255, 1, '', 0),
+(@GUID+113, 181557, 530, 3519, 0, 1, 1, -2839.94, 5601.23, 9.60491, -0.436332, 0, 0, -0.216439, 0.976296, 3600, 255, 1, '', 0),
+(@GUID+115, 181557, 530, 3519, 0, 1, 1, -2611.83, 5453.13, 13.1311, 3.03684, 0, 0, 0, 0, 3600, 255, 1, '', 0),
+(@GUID+117, 181557, 530, 3519, 0, 1, 1, -2515.9, 5468.29, 8.422, -2.60054, 0, 0, -0.96363, 0.267241, 3600, 255, 1, '', 0),
+(@GUID+119, 181557, 530, 3519, 0, 1, 1, -2306.47, 5454.43, 5.74422, -1.76278, 0, 0, 0, 0, 3600, 255, 1, '', 0),
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1666109692456128300.sql:410]:
```suggestion
(@GUID+119, 181557, 530, 3519, 0, 1, 1, -2306.47, 5454.43, 5.74422, -1.76278, 0, 0, 0, 0, 3600, 255, 1, '', 0);
```

File: data/sql/updates/pending_db_world/rev_1666109692456128300.sql [hunk 3]
```diff
@@ -0,0 +1,1310 @@
...
+(@POOL+39, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+40, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+41, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+42, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+43, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+44, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+45, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+46, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+47, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
+(@POOL+48, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein'),
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1666109692456128300.sql:544]:
```suggestion
(@POOL+48, 1, 'Terokkar Forest - Adamantite Deposit / Khorium Vein');
```

File: data/sql/updates/pending_db_world/rev_1666109692456128300.sql [hunk 4]
```diff
@@ -0,0 +1,1310 @@
...
+-- SET @POOL = 12228; -- 24
+
+DELETE FROM `pool_template` WHERE `entry` BETWEEN @POOLMOTHER+0 AND @POOLMOTHER+5;
+INSERT INTO `pool_template` (`entry`, `max_limit`, `description`) VALUES
+(@POOLMOTHER+0, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 12'),
+(@POOLMOTHER+1, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 13'),
+(@POOLMOTHER+2, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 14'),
+(@POOLMOTHER+3, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 15'),
+(@POOLMOTHER+4, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 16'),
+(@POOLMOTHER+5, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 17'),
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1666109692456128300.sql:761]:
```suggestion
(@POOLMOTHER+5, 1, 'Terokkar Forest - Adamantite / Rich / Khorium - Group 17');
```

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1666109692456128300.sql (hunk 4 above)
Comment (Gultask) [reply data/sql/updates/pending_db_world/rev_1666109692456128300.sql]: Thanks! 😛

**Comment by Gultask** [comment]
> > I can't seem to be able to repopulate my DB so I'm out of a local for now.
> 
> Please fetch the upstream of the pr.

Didn't work for me. Worldserver still crashing and dbupdater just doesn't care.

**Comment by acidmanifesto** [comment]
this is interesting. it doesnt like your use of defines for the entries. it sees it as NULL, which is why it is not being accepted by the world.
image

**Comment by acidmanifesto** [comment]
@Kitzunu do you have any ideas what he can do to make his sql passible?

**Comment by Gultask** [comment]
I didn't define anything yet because it's WIP, it's all commented 😛

**Comment by acidmanifesto** [comment]
> I didn't define anything yet because it's WIP, it's all commented 😛
> 
> I need to rebuild the DB to get free entries, but I can't for now.

gotcha.

**Comment by Gultask** [comment]
Gave it another shot and it worked. Building the DB right now. I'll define the entries ASAP 👍

**Review by Kitzunu — APPROVED** [review]

**Review by acidmanifesto — APPROVED** [review]

**Pull request merged**